lib/rucio/core/rule.py

```diff
@@ -674,7 +769,13 @@ def inject_rule(rule_id, *, session: "Session", logger=logging.log):
         # Remove duplicates from the list of dictionaries
         dids = [dict(t) for t in {tuple(d.items()) for d in dids}]
         # Remove dids which already have a similar rule
-        dids = [did for did in dids if session.query(models.ReplicationRule).filter_by(scope=did['scope'], name=did['name'], account=rule.account, rse_expression=rule.rse_expression).count() == 0]
+        stmt = select(
+            models.ReplicationRule
+        ).where(
+            and_(models.ReplicationRule.account == rule.account,
+                 models.ReplicationRule.rse_expression == rule.rse_expression)
+        )
+        dids = [did for did in dids if session.execute(stmt.where(and_(models.ReplicationRule.scope == did['scope'], models.ReplicationRule.name == did['name']))).scalar_one_or_none() is None]
```

Comment:
While this would do the same, I wonder if internally something different happens though in the ORM. 
The original version does a count, thus it only needs to transmit one value from the DB to the client. I wonder if in the `scalar_one_or_none` version more is transmitted, and then this is evaluated client side. Thus more data flows between DB and client (server)

Reply:
From the docs: https://docs.sqlalchemy.org/en/14/core/connections.html#sqlalchemy.engine.Result.scalar_one_or_none 

> Return exactly one scalar result or None. This is equivalent to calling [Result.scalars()](https://docs.sqlalchemy.org/en/14/core/connections.html#sqlalchemy.engine.Result.scalars) and then [Result.one_or_none()](https://docs.sqlalchemy.org/en/14/core/connections.html#sqlalchemy.engine.Result.one_or_none).

I wouldn't expect a significant performance difference between using `count` and `scalar_one_or_none`